### PR TITLE
Reorganize windowmaker dockapps

### PR DESCRIPTION
**Pull request by AndersonTorres**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/window-managers/windowmaker/dockapps/AlsaMixer-app.nix [hunk 1]
```diff
@@ -0,0 +1,30 @@
+{ stdenv, dockapps-sources, pkgconfig, libX11, libXpm, libXext, alsaLib }:
+
+stdenv.mkDerivation rec {
+  pname = "AlsaMixer.app";
+  version = "0.2.1";
+
+  src = dockapps-sources;
+
+  nativeBuildInputs = [ pkgconfig ];
```
Comment (SuperSandro2000) [inline pkgs/applications/window-managers/windowmaker/dockapps/AlsaMixer-app.nix:9]:
```suggestion
  nativeBuildInputs = [ pkg-config ];
```

File: pkgs/applications/window-managers/windowmaker/dockapps/AlsaMixer-app.nix [hunk 2]
```diff
@@ -0,0 +1,30 @@
+{ stdenv, dockapps-sources, pkgconfig, libX11, libXpm, libXext, alsaLib }:
```
Comment (SuperSandro2000) [inline pkgs/applications/window-managers/windowmaker/dockapps/AlsaMixer-app.nix:1]:
```suggestion
{ stdenv, dockapps-sources, pkg-config, libX11, libXpm, libXext, alsaLib }:
```

File: pkgs/applications/window-managers/windowmaker/dockapps/AlsaMixer-app.nix [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+stdenv.mkDerivation rec {
+  pname = "AlsaMixer.app";
+  version = "0.2.1";
+
+  src = dockapps-sources;
+
+  nativeBuildInputs = [ pkgconfig ];
+  buildInputs = [ libX11 libXpm libXext alsaLib ];
+
+  postUnpack = "sourceRoot=\${sourceRoot}/${pname}";
```
Comment (SuperSandro2000) [inline pkgs/applications/window-managers/windowmaker/dockapps/AlsaMixer-app.nix:12]:
```suggestion
  postUnpack = ''
    sourceRoot=\${sourceRoot}/${pname}
  '';
```

File: pkgs/applications/window-managers/windowmaker/dockapps/AlsaMixer-app.nix [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+  dontConfigure = true;
+
+  installPhase = ''
+    runHook preInstall
+    install -d $out/bin
+    install -t $out/bin AlsaMixer.app
+    (cd $out/bin; ln -s AlsaMixer.app AlsaMixer)
```
Comment (SuperSandro2000) [inline pkgs/applications/window-managers/windowmaker/dockapps/AlsaMixer-app.nix:20]:
```suggestion
    ln -s $out/bin/AlsaMixer.app $out/bin/AlsaMixer
```

File: pkgs/applications/window-managers/windowmaker/dockapps/AlsaMixer-app.nix [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+    runHook postInstall
+  '';
+
+  meta = with stdenv.lib; {
+    description = "Alsa mixer application for Windowmaker";
+    homepage = "https://www.dockapps.net/alsamixerapp";
+    license = licenses.gpl2Plus;
+    maintainers = [ maintainers.bstrik ];
```
Comment (SuperSandro2000) [inline pkgs/applications/window-managers/windowmaker/dockapps/AlsaMixer-app.nix:28]:
```suggestion
    maintainers = with maintainers; [ bstrik ];
```

File: pkgs/applications/window-managers/windowmaker/dockapps/default.nix [hunk 6]
```diff
@@ -0,0 +1,21 @@
+{ config, lib, pkgs }:
+
+lib.makeScope pkgs.newScope (self: with self; {
+
+  dockapps-sources = pkgs.fetchgit {
+    url = "https://repo.or.cz/dockapps.git";
+    rev = "b2b9d872ee61c9b329e4597c301e4417cbd9c3ea";
```
Comment (SuperSandro2000) [inline pkgs/applications/window-managers/windowmaker/dockapps/default.nix:7]: Is there a branch or tag we could track?

File: pkgs/applications/window-managers/windowmaker/dockapps/libdockapp.nix [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+  src = dockapps-sources;
+
+  nativeBuildInputs = [ autoreconfHook pkg-config ];
+
+  buildInputs = [ libX11 libXext libXpm fontutil mkfontdir ];
+
+  postUnpack = "sourceRoot=\${sourceRoot}/${pname}";
+
+  # There is a bug on --with-font
+  configureFlags = [ "--with-examples=no" "--with-font=no" ];
```
Comment (SuperSandro2000) [inline pkgs/applications/window-managers/windowmaker/dockapps/libdockapp.nix:18]:
```suggestion
  configureFlags = [
    "--with-examples=no"
    "--with-font=no"
  ];
```

File: pkgs/applications/window-managers/windowmaker/dockapps/libdockapp.nix [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+
+  meta = with stdenv.lib; {
+    description = "A library providing a framework for dockapps";
+    homepage = "https://www.dockapps.net/libdockapp";
+    license = licenses.gpl2Plus;
+    maintainers = [ maintainers.bstrik ];
```
Comment (SuperSandro2000) [inline pkgs/applications/window-managers/windowmaker/dockapps/libdockapp.nix:24]:
```suggestion
    maintainers = with maintainers; [ bstrik ];
```

File: pkgs/applications/window-managers/windowmaker/dockapps/wmCalClock.nix [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+
+  buildInputs = [ libX11 libXpm libXext ];
+
+  postUnpack = "sourceRoot=\${sourceRoot}/${pname}/Src";
+
+  preBuild = ''
+    makeFlagsArray+=( 
+      CC="cc"
+      INCDIR="-I${libX11.dev}/include -I${libXext.dev}/include -I${libXpm.dev}/include"
+      LIBDIR="-I${libX11}/lib -I${libXext}/lib -I${libXpm}/lib" )
```
Comment (SuperSandro2000) [inline pkgs/applications/window-managers/windowmaker/dockapps/wmCalClock.nix:18]:
```suggestion
    makeFlagsArray+=( 
      CC="cc"
      INCDIR="-I${libX11.dev}/include -I${libXext.dev}/include -I${libXpm.dev}/include"
      LIBDIR="-I${libX11}/lib -I${libXext}/lib -I${libXpm}/lib"
    )
```

File: pkgs/applications/window-managers/windowmaker/dockapps/wmsystemtray.nix [hunk 10]
```diff
@@ -1,20 +1,22 @@
...
-  name = "wmsystemtray-1.4";
+stdenv.mkDerivation rec {
+  pname = "wmsystemtray";
+  version = "1.4";
+
   src = fetchurl {
-     url = "mirror://sourceforge/project/wmsystemtray/wmsystemtray/wmsystemtray-1.4.tar.gz";
-     sha256 = "8edef43691e9fff071000e29166c7c1ad420c0956e9068151061e881c8ac97e9";
+    url = "mirror://sourceforge/project/${pname}/${pname}/${pname}-${version}.tar.gz";
+     sha256 = "sha256-jt70NpHp//BxAA4pFmx8GtQgwJVukGgVEGHogcisl+k=";
```
Comment (SuperSandro2000) [inline pkgs/applications/window-managers/windowmaker/dockapps/wmsystemtray.nix:9]:
```suggestion
    url = "mirror://sourceforge/project/${pname}/${pname}/${pname}-${version}.tar.gz";
    sha256 = "jt70NpHp//BxAA4pFmx8GtQgwJVukGgVEGHogcisl+k=";
```

File: pkgs/applications/window-managers/windowmaker/dockapps/default.nix (hunk 6 above)
Comment (AndersonTorres) [reply pkgs/applications/window-managers/windowmaker/dockapps/default.nix:7]: Not so easily. It is essentially a monorepo of a plethora of dockapps.

File: pkgs/applications/window-managers/windowmaker/dockapps/wmsystemtray.nix (hunk 10 above)
Comment (AndersonTorres) [reply pkgs/applications/window-managers/windowmaker/dockapps/wmsystemtray.nix:9]: Well, this is the format spit out by the command when I use it via fakeSha256 trick.

**Pull request merged**